### PR TITLE
test error type for dict contructor, #18708

### PR DESCRIPTION
**Pull request by KristofferC**
Closes https://github.com/JuliaLang/julia/issues/18708.
Closes https://github.com/JuliaLang/julia/issues/19179

### CODE REVIEW

**Comment by KristofferC** [comment]
Added test for https://github.com/JuliaLang/julia/issues/19179

File: test/dict.jl [hunk 1]
```diff
@@ -662,3 +662,11 @@ end
     @test f.x[2].first == (-, 6, 5, 3, 5, 8)
     @test f.x[2].second == 1
 end
+
+@testset "issue #18708 error type for dict constructor" begin
+    @test_throws UndefVarError Dict(x => y for x in 1:10)
+end
+
+@testset "issue #19179 throwing error in dict constructor" begin
+    @test_throws ErrorException Dict(i => error("foo") for i in 1:10)
```
Comment (tkelman) [inline test/dict.jl]: would have less chance of this passing unintentionally if it were throwing and testing for a very specific exception type defined just for the test

Comment (KristofferC) [inline test/dict.jl]: I think this is OK though.

Comment (tkelman) [inline test/dict.jl]: if the error that the issue was complaining about were an ErrorException (or changed to one) then this test wouldn't identify the problem

Comment (KristofferC) [inline test/dict.jl]: fine updated

**Comment by KristofferC** [comment]
bump

**Comment by KristofferC** [comment]
Dict-test passed, error on travis is

```
Error in testset spawn:
Error During Test
  Got an exception of type LoadError outside of a @test
  LoadError: InterruptException:
  while loading /private/tmp/julia/share/julia/test/spawn.jl, in expression starting on line 70
```

**Comment by JeffBezanson** [comment]
I already added a test for this in df01d345103ed351548c1280f791211bb8b8ac33, but I didn't know there was an issue about it.

**Comment by tkelman** [comment]
better to have 2 tests for the same thing than 0, as long as they don't individually take long to run

**Pull request merged**